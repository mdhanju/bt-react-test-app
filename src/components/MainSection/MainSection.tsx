import React, { ReactNode } from 'react'
import classnames from 'classnames'

const MainSection = ({
    className,
    children,
    ...props
}: Props): React.ReactElement => (
    <section className={classnames(className)} {...props}>
        {children}
    </section>
)

type Props = {
    className?: string;
    children?: ReactNode;
};

export default MainSection
