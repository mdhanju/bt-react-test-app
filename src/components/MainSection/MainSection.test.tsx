import React from 'react'

import MainSection from './MainSection'

describe('<MainSection />', (): void => {
    it('should render properly without crashing', () => {
        const wrapper = shallow(<MainSection />)
        expect(wrapper.exists()).toBe(true)
        expect(wrapper.getElements()).toMatchSnapshot()
    })

    it('render the section with child div', () => {
        const wrapper = render(
            <MainSection>
                <div>Help</div>
            </MainSection>
        )
        expect(wrapper.html()).toMatch('<div>Help</div>')
    })

    it('render the section with child Component', () => {
        const ChildComp = (): React.ReactElement => {
            return <div>Child 1</div>
        }
        const wrapper = render(
            <MainSection>
                <ChildComp />
            </MainSection>
        )
        expect(wrapper.html()).toMatch('<div>Child 1</div>')
    })
})
