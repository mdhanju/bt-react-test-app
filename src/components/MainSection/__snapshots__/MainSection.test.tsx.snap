// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainSection /> should render properly without crashing 1`] = `
Array [
  <section
    className=""
  />,
]
`;
